### DISCUSSION TITLE
Password lost, no recovery email

### DISCUSSION THREAD

**Discussion by Simon248** [firefly-iii/firefly-iii#7928] [Firefly III] [answered]
Hello, 

Thanks for taking time to help me! I hope that my description will be clear enough.

As in the tittle, I've lost my password.
Firefly iii is running in docker and i'm the only user.
I didn't activate mail recovery.
I can go to the login page and to the /password/reset page.

I followed this https://github.com/firefly-iii/firefly-iii/issues/1611#issuecomment-413237987
But i get this error in docker logs and in firefly logs (under /storage/logs):
"Cowardly refuse to send a password reset message to user #1 because the reset button has been throttled."
So maybe I desactivated the reset password button.

I looked at the database to modify this value and try different value for the field "reset" in table "users". I tried ("activated"; "active"; "1"; "true") but always the same problem.

I observed that when I try to recover my password via  /password/reset page : 
-on the first time nothing happen in the logs and an entry is created in database in the table password_reset. (so I can get the hached token but dont have the not-hached token in logs)
-on the second time, I've got a new file created in logs: ff3-cli-<date>.log with the error message : "Cowardly refuse to send a password reset message to user #1 because the reset button has been throttled."

I don't understand why i can't have the recovery link in the logs.

Attached my env settings for log.

[dotenv.txt](https://github.com/firefly-iii/firefly-iii/files/12514370/dotenv.txt)

**Comment by JC5** [comment, marked as answer]
https://docs.firefly-iii.org/firefly-iii/faq/other/#i-lost-my-password-but-resetting-doesnt-work

**Comment by Simon248** [comment]
Thanks, it lead me to the answer. I forgot that environement variable dont survive a docker restart.
BTW : thanks for your job.
